core/test/validation_test.jl

```diff
@@ -49,30 +49,45 @@ end
 end
 
 @testitem "Neighbor count validation" begin
-    using Graphs: DiGraph, add_edge!
+    using MetaGraphsNext: MetaGraph
+    using Graphs: DiGraph
     using Logging
 
-    graph_flow = DiGraph(6)
-    add_edge!(graph_flow, 2, 1)
-    add_edge!(graph_flow, 3, 1)
-    add_edge!(graph_flow, 6, 2)
+    graph = MetaGraph(
+        DiGraph();
+        label_type = Ribasim.NodeID,
+        vertex_data_type = Ribasim.NodeMetadata,
+        edge_data_type = Ribasim.EdgeMetadata,
+        graph_data = nothing,
+    )
+
+    for i in 1:6
+        type = i in [1, 6] ? :pump : :other
+        graph[Ribasim.NodeID(i)] = Ribasim.NodeMetadata(type, 9)
+    end
 
-    graph_control = DiGraph(6)
-    add_edge!(graph_control, 5, 6)
+    graph[Ribasim.NodeID(2), Ribasim.NodeID(1)] =
+        Ribasim.EdgeMetadata(Ribasim.EdgeID(1), Ribasim.EdgeType.flow)
```

Comment:
It probably makes sense to add `using Ribasim: ...` for these ID and metadata types at the top of this testitem.